config/routes.rb

```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  get 'users', to: 'users#index', as: 'user_index'
+  get 'users/:id', to: 'users#show', as: 'user_show'
```

Comment:
- [ ] **[OPTIONAL]** Good job exposing only the needed routes, but I think we can save some code here with the `resources` helper:

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```